### PR TITLE
Adding tutorial for user defined bases

### PR DESCRIPTION
**Pull request by bsmith24**

### CODE REVIEW

File: step3/user_defined_basis/hot_electron_cooling/build_bases.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+"""
+ 3. Compute overlaps and time-overlaps between the single-particle excitations 
+    3.1. Compute overlaps in the basis of single-particle excitations
+    3.2. Compute time-overlaps in the basis of single-particle excitations
+    3.3. Convert the data types to Libra's CMATRIX
+"""
+#######################################################################################
+
+# 3.1. Overlaps
+pool = mp.Pool( 24 )
```
Comment (d-b-w) [inline step3/user_defined_basis/hot_electron_cooling/build_bases.py:127]: Prefer to use the context manager:

    with mp.Pool(24) as pool:
        tmp_s_sd = pool.map(...

**Comment by MohammadShakiba** [comment]
Hi Daniel, Thanks for your comment on this. We'll consider using this not only here but also in the rest of the Libra code too. Thanks again.

**Comment by d-b-w** [comment]
Cool! Sorry for the drive-by review :)

**Comment by bsmith24** [comment]
Hi Daniel. We greatly appreciate your feedback!

**Pull request merged**